MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -45,56 +38,80 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupID:(NSString *)groupID {
   }
   NSData *logData = [NSKeyedArchiver archivedDataWithRootObject:log];
   NSString *base64Data = [logData base64EncodedStringWithOptions:NSDataBase64EncodingEndLineWithLineFeed];
-  NSString *addLogQuery = [NSString stringWithFormat:@"insert or replace into %@ values ('%@', '%@')",
-                           kMSLogTableName, groupID, base64Data];
+  NSString *addLogQuery = [NSString
+      stringWithFormat:@"insert or replace into %@ values ('%@', '%@')", kMSLogTableName, groupID, base64Data];
   return [self.connection executeQuery:addLogQuery];
 }
 
 - (NSArray<MSLog> *)deleteLogsForGroupID:(NSString *)groupID {
-  NSArray<MSLog> *logs = [self getLogsWith:groupID];
-  [self deleteLogsWith:groupID];
+  NSArray<MSLog> *logs = [self getLogsWithGroupID:groupID];
+  [self deleteLogsWithGroupID:groupID];
   return logs;
 }
 
-- (void)deleteLogsForId:(NSString *)logsId withGroupID:(NSString *)groupID {
+- (void)deleteLogsForId:(__attribute__((unused)) NSString *)batchId withGroupID:(NSString *)groupID {
 
-  // FIXME: logsId ?
-  [self deleteLogsWith:groupID];
+  // FIXME: Restore batch deletion.
+  [self deleteLogsWithGroupID:groupID];
 }
 
-- (BOOL)loadLogsForGroupID:(NSString *)groupID withCompletion:(nullable MSLoadDataCompletionBlock)completion {
-  NSMutableArray<MSLog> *logs = [self getLogsWith:groupID];
-
-  // Remove excess logs
-  if (logs.count > self.bucketFileLogCountLimit) {
-    [logs removeObjectsInRange:NSMakeRange(self.bucketFileLogCountLimit, logs.count - self.bucketFileLogCountLimit)];
+- (BOOL)loadLogsForGroupID:(NSString *)groupID
+                     limit:(NSUInteger)limit
+            withCompletion:(nullable MSLoadDataCompletionBlock)completion {
+
+  /*
+   * There is a need to determine if there will be more logs available than those under the limit.
+   * So this is just about knowing if there is at least 1 log above the limit.
+   * Thus, just 1 log is added to the requested limit and then removed later as needed.
+   */
+  NSMutableArray<MSLog> *logs = (NSMutableArray<MSLog> *)[self getLogsWithGroupID:groupID limit:(limit + 1)];
```

Comment:
Do we really need this one? I think it is okay to let LogManager handles based on channel configuration.
Based on channel configuration, it will automatically pick up another batch if it has more than batch size.